File: frontend/src/types/service-options.ts
export interface ServiceOption {
  id: string;
  type: string;
  label: string;
  description: string;
  fee: number | null;
  available: boolean;
}

export interface ServiceOptionsResponse {
  floor_options: ServiceOption[];
  ladder_options: ServiceOption[];
  special_vehicle_options: ServiceOption[];
}

export interface SelectedOptions {
  floor_option_id?: string;
  ladder_option_id?: string;
  special_vehicle_option_id?: string;
  total_fee: number;
} 

File: backend/app/schemas/service_options.py
from pydantic import BaseModel
from typing import Optional, List

class ServiceOption(BaseModel):
    id: str
    type: str  # 'floor', 'ladder', 'special'
    label: str
    description: str
    fee: Optional[int] = None
    available: bool

    class Config:
        from_attributes = True

class ServiceOptionsResponse(BaseModel):
    floor_options: List[ServiceOption]
    ladder_options: List[ServiceOption]
    special_vehicle_options: List[ServiceOption]

    class Config:
        from_attributes = True

class SelectedOptions(BaseModel):
    floor_option_id: Optional[str] = None
    ladder_option_id: Optional[str] = None
    special_vehicle_option_id: Optional[str] = None
    total_fee: int

    class Config:
        from_attributes = True